### PR TITLE
feat(completion): add supermaven-nvim

### PR DESCRIPTION
**Pull request by taskylizard**
## 📑 Description

Adds configuration for https://github.com/supermaven-inc/supermaven-nvim.

### CODE REVIEW

File: lua/astrocommunity/completion/supermaven-nvim/init.lua
Comment (Uzaaft) [inline lua/astrocommunity/completion/supermaven-nvim/init.lua:1]: Does the keymaps appear in which key?

Comment (Uzaaft) [reply lua/astrocommunity/completion/supermaven-nvim/init.lua:1]: Ah true, they are ctrl based.

File: lua/astrocommunity/completion/supermaven-nvim/init.lua [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    require("supermaven-nvim").setup {
+      keymaps = {
+        accept_suggestion = "<C-l>",
+        clear_suggestion = "<C-h>",
+        accept_word = "<C-w>",
+      },
+      log_level = "warn",
+      disable_inline_completion = false, -- disables inline completion for use with cmp
+      disable_keymaps = false, -- disables built in keymaps for more manual control
+    }
```
Comment (Uzaaft) [inline lua/astrocommunity/completion/supermaven-nvim/init.lua:14]: These can be moved to an opts instead.

Comment (taskylizard) [reply lua/astrocommunity/completion/supermaven-nvim/init.lua]: Indeed yes, I just followed the documentation for this but this is better, fixed.

**Review by Uzaaft — APPROVED** [review]

**Pull request merged**